frontend/src/features/drug_discovery/components/RunDockingStep.vue

```diff
@@ -150,18 +150,61 @@
     </q-table>
 </template>
 
-<script>
-import { useDrugDiscoveryStore } from 'src/features/drug_discovery/storage';
+<script lang="ts">
+import {useDrugDiscoveryStore} from 'src/features/drug_discovery/storage';
 import {useRoute} from "vue-router";
 import PdbViewer from "src/components/PdbViewer.vue";
-
-export default {
+import {ResultMetaData} from "src/api/client/models/ResultMetaData.ts";
+import {defineComponent} from "vue";
+
+interface Job {
+  job_id: string;
+  target_name: string | undefined;
+  ligand_name: string | undefined;
+  target_id: string;
+  ligand_id: string;
+  progress?: number;
+  isRunning?: boolean;
+  msaAvailable?: boolean;
+  pocketAvailable?: boolean;
+  foldingAvailable?: boolean;
+  dockingAvailable?: boolean;
+  msaRunning?: boolean;
+  pocketRunning?: boolean;
+  foldingRunning?: boolean;
+  dockingRunning?: boolean;
+  pocketIds: number[] | null;
+  isAnyJobRunning: boolean;
+  [key: string]: any; // Add this line to allow dynamic keys
+}
+
+interface ResultData {
+  predicted_pdb?: string;
+  predicted_sdf?: string;
+  predicted_pdb_file?: File;
+  predicted_sdf_file?: File;
+}
+
+interface DockingResult {
+  target_name: string | undefined;
+  ligand_name: string | undefined;
+  resultData?: ResultData | null;
+  pocketIds: number[] | null;
+  experiment_id: string;
+  job_id: string;
+  target_id: string;
+  ligand_id: string;
+  expanded: boolean;
+}
+
+export default defineComponent({
   name: 'RunDocking',
   components: {PdbViewer},
   data() {
     return {
-      dockingResults: [],
-      jobsInQueue: [],
+      experimentId: '' as string,
```

Comment:
Here it is better to use null, '' looks like a 'magic value' for undefined

Reply:
Addressed in next commit